### PR TITLE
chore(sql): support PostgreSQL `insert` ... `with` syntax

### PR DESCRIPTION
**Pull request by bluestreak01**
Follow up for https://github.com/questdb/questdb/pull/2724 to implement PostgreSQL `insert` syntax:

https://www.postgresql.org/docs/current/sql-insert.html

### CODE REVIEW

File: core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java [hunk 1]
```diff
@@ -782,7 +782,7 @@ public void testEqGeoHashWhenOtherIsStr2() throws Exception {
     @Test
     public void testFailsForLatestByOnSubQueryWithNoTimestampSpecified() throws Exception {
         assertFailure(
-                "with tab as (x where b in ('BB')) tab latest by b",
+                "with tab as (x where b in ('BB')) select * from tab latest by b",
```
Comment (amyshwang) [inline core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java:785]:
```suggestion
                "with tab as (x where b in ('BB')) select * from tab latest on b",
```
I don't think this matters for this PR but probably better to remove latest by?

Comment (bluestreak01) [reply core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java:785]: this is a "negative" test, which verifies behaviour on invalid SQL

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|6|6|100.00%|

**Review by bziobrowski — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|6|6|100.00%|

**Pull request merged**